### PR TITLE
op: Implement stencil_read_write_mask test in stencil.spec.ts

### PR DESCRIPTION
**Pull request by Gyuyoung**
This PR adds a new test to check if the stencil read/write masks work correctly.

Issue: #2023

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [x] Tests are properly located in the test tree.
- [x] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [x] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [x] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/api/operation/rendering/stencil.spec.ts [hunk 1]
```diff
@@ -230,3 +230,65 @@ g.test('stencil_compare_func')
...
+    } as const;
+
+    const baseState = {
+      format: depthSpencilFormat,
+      depthWriteEnabled: false,
+      depthCompare: 'always',
+      stencilFront: baseStencilState,
+      stencilBack: baseStencilState,
+      stencilReadMask: 0xff,
+      stencilWriteMask: maskType === 'write' ? 0x1 : 0xff,
```
Comment (kainino0x) [inline src/webgpu/api/operation/rendering/stencil.spec.ts:280]: The test will still pass if you remove this (simulating a bug where stencilWriteMask is ignored). This mask is applied to a stencil write value of 1 (hardcoded in runStencilStateTest) so it has no effect.

File: src/webgpu/api/operation/rendering/stencil.spec.ts [hunk 2]
```diff
@@ -230,3 +230,65 @@ g.test('stencil_compare_func')
...
+    } as const;
+
+    const state = {
+      format: depthSpencilFormat,
+      depthWriteEnabled: false,
+      depthCompare: 'always',
+      stencilFront: stencilState,
+      stencilBack: stencilState,
+      stencilReadMask: maskType === 'read' ? 0x1 : 0xff,
```
Comment (kainino0x) [inline src/webgpu/api/operation/rendering/stencil.spec.ts:289]: The test will also still pass if you remove this (simulating a bug where stencilReadMask is ignored). The reason is similar: the stencil attachment is cleared to 1 by runStencilStateTest, so the mask has no effect.

File: src/webgpu/api/operation/rendering/stencil.spec.ts (hunk 1 above)
Comment (Gyuyoung) [reply src/webgpu/api/operation/rendering/stencil.spec.ts:364]: Ah, right. then, it's likely this PR needs to have a way to set the stencil reference value of the base state. So, I update runStencilStateTest in order to make the base state have its own stencil value. Then, I set the stencil reference value of the base state to 3.

File: src/webgpu/api/operation/rendering/stencil.spec.ts (hunk 2 above)
Comment (Gyuyoung) [reply src/webgpu/api/operation/rendering/stencil.spec.ts]: ditto.

File: src/webgpu/api/operation/rendering/stencil.spec.ts [hunk 3]
```diff
@@ -310,12 +298,87 @@ g.test('stencil_fail_operation')
...
-      { state: passState, color: kGreenStencilColor },
+      { state: passState, color: kGreenStencilColor, stencil: stencilRefValue },
+    ];
+    t.runStencilStateTest(testStates, _expectedColor);
+  });
+
+g.test('stencil_read_write_mask')
+  .desc(
+    `
+  Tests that setting a stencil read/write masks work. Basically, The base triangle sets 1 to the
```
Comment (kainino0x) [inline src/webgpu/api/operation/rendering/stencil.spec.ts:316]:
```suggestion
  Tests that setting a stencil read/write masks work. Basically, The base triangle sets 3 to the
```

**Comment by Gyuyoung** [comment]
> LGTM! BTW please use clean merge commits where possible, instead of rebase, to make it easier for me to review incremental PR changes. (In this case, on the branch we would have 3 commits: the original commit I reviewed, then a clean merge commit I can ignore, then your new changes)

Ah, I'm sorry. It's likely it happened I'm requesting some PRs simultaneously. Then, is it better to request to review a PR that has multiple commits?

**Pull request merged**